### PR TITLE
[Enhancement] Let instrucotrs set student exams to submitted/unsubmitted

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

With this change instructors can set the` submitted` state of individual student exams. E.g.: if a student did not manage to submit the exam in time and the instructor decides the exam should count nonetheless, he can set the exam to submitted by himself.

### Description

In the student-exam-detail following things were changed:
1. Instructors can see the Matriculation number now.
2. I added a "Change Student Exam to Submitted" button, where the submission status can be  toggled.

If the exam was not submitted:

If it was submitted, but the instructor wants to set it to "unsubmitted":

### Steps for Testing

1. create an exam and start the participation with a student. Don't hand  in the exam.
2. After the exam is over go to exam -> student-exams -> View (of the exam which you did start, but not hand in)
3. Use the "Change Student Exam to Submitted" button.
4. Go to the student account and check if the now handed in exam is visible.

--->

### CODE REVIEW

**Review by JohannesWeiss — DISMISSED** [review]
Tested on ts1, worked as expected.

**Review by simonzilker — DISMISSED** [review]
Works as described!

**Review by marvinchimeka — APPROVED** [review]
Works as expected. No issues found

**Review by LudwigWeide — APPROVED** [review]
Works as expected.

**Review by airelawaleria — APPROVED** [review]

**Review by airelawaleria — DISMISSED** [review]

**Review by tiffca — APPROVED** [review]
Tested successfully

**Review by EneaGore — APPROVED** [review]
All seems good on ts1.

File: src/main/webapp/app/course/manage/course-detail.component.html [hunk 1]
```diff
@@ -50,7 +50,7 @@ <h4 class="modal-title"><span [jhiTranslate]="'artemisApp.course.archive.popup.t
                     <div class="modal-body">
                         <p [jhiTranslate]="'artemisApp.course.archive.popup.question'" [translateValues]="{ title: course.title }">Are you sure you want to archive the course?</p>
                         <p [jhiTranslate]="'artemisApp.course.archive.popup.statement1'">
-                            The process will compress all student code repositories, file upload exercises, modeling exercises, and text exercises for exercises and exams.
+                            The process will compggggress all student code repositories, file upload exercises, modeling exercises, and text exercises for exercises and exams.
```
Comment (nicolasruscher) [inline src/main/webapp/app/course/manage/course-detail.component.html:53]: I think this is a typo? :D

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts [hunk 2]
```diff
@@ -11,13 +11,16 @@ import { ArtemisDurationFromSecondsPipe } from 'app/shared/pipes/artemis-duratio
...
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ButtonSize } from 'app/shared/components/button.component';
 
 @Component({
     selector: 'jhi-student-exam-detail',
     templateUrl: './student-exam-detail.component.html',
     providers: [ArtemisDurationFromSecondsPipe],
 })
 export class StudentExamDetailComponent implements OnInit {
+    readonly ButtonSize = ButtonSize;
```
Comment (nicolasruscher) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts:23]: Is this really needed? I could not found a usage within the html or code :)

**Review by nicolasruscher — APPROVED** [review]
I only looked over the code, but apart from some minor things I did not find any problems, looks like a very feature!

File: src/main/webapp/app/course/manage/course-detail.component.html (hunk 1 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/course/manage/course-detail.component.html:53]: lol yes
```suggestion
                            The process will compress all student code repositories, file upload exercises, modeling exercises, and text exercises for exercises and exams.
```
I guess my 'g' key got stuck

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts (hunk 2 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts]: I removed it

**Review by marvinchimeka — APPROVED** [review]
Re-approve

**Review by JohannesWeiss — APPROVED** [review]
Re-approve

**Review by simonzilker — APPROVED** [review]
Re-approve

**Review by airelawaleria — APPROVED** [review]

**Review by philgabr — APPROVED** [review]
Approve newest build (8)

**Review by krusche — APPROVED** [review]
Code looks good. Thanks for this nice contribution 👍

**Pull request merged**